controller/api/destination/server.go

```diff
@@ -264,17 +267,27 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 /// util ///
 ////////////
 
-func nsFromToken(token string) string {
-	// ns:<namespace>
-	parts := strings.Split(token, ":")
-	if len(parts) == 2 && parts[0] == "ns" {
-		return parts[1]
-	}
+type contextToken struct {
+	Ns       string `json:"ns,omitempty"`
+	NodeName string `json:"nodeName,omitempty"`
+}
 
-	return ""
+func (s *server) parseContextToken(token string) contextToken {
+	ctxToken := contextToken{}
+	if err := json.Unmarshal([]byte(token), &ctxToken); err != nil {
+		// if json is invalid, means token can have ns:<namespace> form
+		parts := strings.Split(token, ":")
+		if len(parts) == 2 && parts[0] == "ns" {
+			s.log.Warnf("context token %s using old token format", token)
+			ctxToken.Ns = parts[1]
```

Comment:
This is probably fine but I don't like relying on the assumption that when `Unmarshal` returns `err` that it won't have written anything to `ctxToken`.  i.e. I don't like assuming that `ctxToken.NodeName` is still the empty value.  This is probably paranoia, but to be super safe I prefer:

```
ctxToken = contextToken {
    Ns: parts[1],
}
```

Reply:
@adleong thank you, that makes a lot of sense, I've made all necessary changes.

I prefer a more defensive programming style, eliminates some of the work that might have to be done in the future to fix issues that slip through the cracks. The suggestion is very appreciated!